tide-whisperer.go

```diff
@@ -99,6 +117,23 @@ func main() {
 		return !(perms["root"] == nil && perms["view"] == nil)
 	}
 
+	//log error detail and write as application/json
+	jsonError := func(res http.ResponseWriter, err detailedError, startedAt time.Time) {
+
+		if err.RawMessage == "" {
```

Comment:
Oh, you could do that.  I have no problem with keeping the RawMessage as part of the error object.  Might want to call it "InternalData"/"InternalMessage"/"LogMessage" to indicate it is not sent back.  I was just saying that there wasn't any point of copying the Message to RawMessage if you also logged Message.

So,

`log.Println(DATA_API_PREFIX, fmt.Sprintf("[%s] [%s] failed after [%.5f]secs with error [%s][%s] ", err.Id, err.Code, time.Now().Sub(startedAt).Seconds(), err.Message, err.RawMessage))`